### PR TITLE
Run end-to-end tests in default workflow

### PR DESCRIPTION
**Pull request by weilbith**
Closes: #2586

As always I recommend to review commit by commit. This was an iterative process, as I needed to test it again and again with CircleCI. Though I already had to rebase some fixup commits. Details in the messages.

PS: The time improvements got postponed for the moment as they where lacking some features to be productive ready. As there are better improvements planned this does not get finished now.

### CODE REVIEW

**Comment by weilbith** [comment]
Times:

- `18:23` for current `master`, note that the SDK unit tests failed, but should affect total time ([CircleCI run](https://app.circleci.com/pipelines/github/raiden-network/light-client/8572/workflows/eca9c07e-8502-45f6-b19b-c03b88c638a6))

File: e2e-environment/README.md [hunk 1]
```diff
@@ -41,13 +45,13 @@ docker run --detach --rm \
...
   --publish 127.0.0.1:8545:8545 \
-  raidennetwork/lightclient-e2e-environment \
+  raidennetwork/lightclient-e2e-environment
 ```
 
 ## Access Services
 
-After starting the Docker container as described above, the services are
-accessible at the following ports on `localhost`:
+After the the Docker container go started, the services are accessible at the
```
Comment (andrevmatos) [inline e2e-environment/README.md:53]: `s/the the/the/`
`s/go started/get started/`

**Review by andrevmatos — APPROVED** [review]
I think I didn't notice where did you update the raiden version to 1.2 to get the `test_dapp_e2e` passing, but the CI changes are great, good job there!

File: e2e-environment/README.md (hunk 1 above)
Comment (weilbith) [reply e2e-environment/README.md:53]: Haha, you know that GitHub supports review suggestions? But copy pasting the commands to Vim is also nice. :rofl:

**Comment by weilbith** [comment]
> I think I didn't notice where did you update the raiden version to 1.2 to get the test_dapp_e2e passing, but the CI changes are great, good job there!

This wasn't part of this PR. And apparently, for some reason the dApp end-to-end tests do work again. I honestly don't know why. But somehow it wasn't the PC causing the issues?

File: e2e-environment/README.md (hunk 1 above)
Comment (weilbith) [reply e2e-environment/README.md]: :heavy_check_mark:

**Pull request merged**